zookeeper/core/component.py

```diff
@@ -354,6 +381,21 @@ def __component_init__(instance, **kwargs):
                 "Keyword arguments passed to component `__init__` must correspond to "
                 f"component fields. Received non-matching argument '{name}'."
             )
+        if utils.is_component_instance(value):
+            if value.__component_configured__:
+                raise ValueError(
+                    "Sub-component instances passed to the `__init__` method of a "
+                    "component must not already be configured. Received configured "
+                    f"component argument '{name}={repr(value)}'."
+                )
```

Comment:
I think it can happen very easily, because you can pass keyword args into the component at instantiation. E.g. the following would trigger the error message:
```python
@component
class Child:
    a: int = Field()

@component
class Parent:
    child: Child = ComponentField()

child = Child()
configure(child, {"a": 42})
parent = Parent(child=child)
```

Reply:
Ah I see, yeah makes sense :+1: